src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt

```diff
@@ -5,87 +5,94 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.Pris
+  }
+
+  private fun isPersonParoleEligible(): EligibilityCheck = early@{
+    if (it.paroleEligibilityDate != null) {
+      if (it.paroleEligibilityDate.isAfter(LocalDate.now(clock))) {
+        return@early true
       }
     }
-    return false
+    return@early false
   }
 
-  private fun PrisonerSearchPrisoner.hasCorrectLegalStatus(): Boolean {
-    return this.legalStatus != "DEAD"
-  }
+  private fun hasCorrectLegalStatus(): EligibilityCheck = { it.legalStatus != "DEAD" }
 
-  private fun PrisonerSearchPrisoner.isOnIndeterminateSentence(): Boolean {
-    return this.indeterminateSentence
-  }
+  private fun isOnIndeterminateSentence(): EligibilityCheck = { it.indeterminateSentence }
 
-  private fun PrisonerSearchPrisoner.hasConditionalReleaseDate(): Boolean {
-    return this.conditionalReleaseDate != null
-  }
+  private fun hasConditionalReleaseDate(): EligibilityCheck = { it.conditionalReleaseDate != null }
 
-  private fun PrisonerSearchPrisoner.isEligibleIfOnAnExtendedDeterminateSentence(): Boolean {
+  private fun isEligibleIfOnAnExtendedDeterminateSentence(): EligibilityCheck = early@{
     // If you don’t have a PED, you automatically pass this check as you’re not an EDS case
-    if (this.paroleEligibilityDate == null) {
-      return true
+    if (it.paroleEligibilityDate == null) {
+      return@early true
     }
 
     // if ARD is not between CRD - 4 days and CRD inclusive (to account for bank holidays and weekends), not eligible
-    if (this.confirmedReleaseDate != null) {
-      val dateStart = this.conditionalReleaseDate!!.minusDays(4)
-      if (this.confirmedReleaseDate.isBefore(dateStart) || this.confirmedReleaseDate.isAfter(this.conditionalReleaseDate)) {
-        return false
+    if (it.confirmedReleaseDate != null) {
+      val dateStart = it.conditionalReleaseDate!!.minusDays(4)
+      if (it.confirmedReleaseDate.isBefore(dateStart) || it.confirmedReleaseDate.isAfter(it.conditionalReleaseDate)) {
+        return@early false
       }
     }
 
     // an APD with a PED in the past means they were a successful parole applicant on a later attempt, so not eligible
-    if (this.actualParoleDate != null) {
-      return false
+    if (it.actualParoleDate != null) {
+      return@early false
     }
 
-    return true
+    return@early true
   }
 
-  private fun PrisonerSearchPrisoner.hasActivePrisonStatus() = status?.let {
-    it.startsWith("ACTIVE") || it == "INACTIVE TRN"
-  } ?: false
-
-  private fun PrisonerSearchPrisoner.hasEligibleReleaseDate(): Boolean {
-    val releaseDate = this.confirmedReleaseDate ?: this.conditionalReleaseDate
+  private fun hasActivePrisonStatus(): EligibilityCheck = { prisoner ->
+    prisoner.status?.let {
+      it.startsWith("ACTIVE") || it == "INACTIVE TRN"
+    } ?: false
+  }
 
-    return releaseDate!!.isEqual(LocalDate.now(clock)) || releaseDate.isAfter(LocalDate.now(clock))
+  private fun hasReleaseDateInTheFuture(): EligibilityCheck = early@{
```

Comment:
I renamed this from `hasEligibleReleaseDate` to  `hasReleaseDateInTheFuture` and made it not fire if it was missing a release date - so that could be picked up by a different rule

Reply:
may be we need another rule for hasNoReleaseDate?

Reply:
If they don't have a CRD, they would be ineligible and fail that earlier check and so be ineligible so is there a reason to check the other dates if they don't have the main CRD date?